component/prometheus/exporter/exporter.go

```diff
@@ -133,16 +137,19 @@ func newExporter(creator Creator, name string, targetBuilderFunc func(discovery.
 		}
 		httpData := data.(http_service.Data)
 
+		componentName := opts.ID[:strings.LastIndex(opts.ID, ".")]
```

Comment:
What is the reasoning for this change just for `unix`? Can you add that to the PR description?

Reply:
The reason is that `prometheus.exporter.unix` doesn't support labels, so we cannot correctly set the component name and ID labels correctly. 

I'll write this on the PR description so it's clearer.